src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -91,165 +106,271 @@ def _load_refcase(self, refcase_file: Optional[str]) -> Optional[EclSum]:
+            base_surface = options.get(ConfigKeys.BASE_SURFACE_KEY)
+            forward_int = _str_to_bool(options.get(ConfigKeys.FORWARD_INIT, "0"))
+
+        return EnkfConfigNode.create_surface(
+            name,
+            init_file,
+            out_file,
+            base_surface,
+            forward_int,
+        )
+
+    @staticmethod
+    def get_field_node(
+        field: Union[dict, list], grid: Optional[EclGrid], trans_table: Any
+    ) -> EnkfConfigNode:
+        if isinstance(field, dict):
+            name = field.get(ConfigKeys.NAME)
+            var_type = field.get(ConfigKeys.VAR_TYPE)
+            out_file = field.get(ConfigKeys.OUT_FILE)
+            enkf_infile = field.get(ConfigKeys.ENKF_INFILE)
+            forward_init = field.get(ConfigKeys.FORWARD_INIT)
+            init_transform = field.get(ConfigKeys.INIT_TRANSFORM)
+            output_transform = field.get(ConfigKeys.OUTPUT_TRANSFORM)
+            input_transform = field.get(ConfigKeys.INPUT_TRANSFORM)
+            min_ = field.get(ConfigKeys.MIN_KEY)
+            max_ = field.get(ConfigKeys.MAX_KEY)
+            init_files = field.get(ConfigKeys.INIT_FILES)
+        else:
+            name = field[0]
+            var_type = field[1]
+            out_file = field[2]
+            enkf_infile = None
+            options = _option_dict(field, 2)
+            if var_type == ConfigKeys.GENERAL_KEY:
+                enkf_infile = field[3]
+            forward_init = _str_to_bool(options.get(ConfigKeys.FORWARD_INIT, "0"))
+            init_transform = options.get(ConfigKeys.INIT_TRANSFORM)
+            output_transform = options.get(ConfigKeys.OUTPUT_TRANSFORM)
+            input_transform = options.get(ConfigKeys.INPUT_TRANSFORM)
+            min_ = options.get(ConfigKeys.MIN_KEY)
+            max_ = options.get(ConfigKeys.MAX_KEY)
+            init_files = options.get(ConfigKeys.INIT_FILES)
+        return EnkfConfigNode.create_field(
+            name,
+            var_type,
+            grid,
+            trans_table,
+            out_file,
+            enkf_infile,
+            forward_init,
+            init_transform,
+            output_transform,
+            input_transform,
+            min_,
+            max_,
+            init_files,
+        )
+
+    @classmethod
+    def _get_schedule_file_node(
+        cls, schedule_file: Union[dict, list, str], tag_format: str
+    ) -> EnkfConfigNode:
+        if isinstance(schedule_file, dict):
+            file_path = schedule_file.get(ConfigKeys.TEMPLATE)
+            file_name = _get_filename(file_path)
+            parameter = schedule_file.get(ConfigKeys.PARAMETER_KEY)
+            init_files = schedule_file.get(ConfigKeys.INIT_FILES)
+        else:
+            file_path = schedule_file[0]
+            if isinstance(schedule_file, str):
+                file_path = schedule_file
+            file_name = _get_filename(file_path)
+            options = _option_dict(file_path, 1)
```

Comment:
yes indeed - we don't have good coverage here yet, as the config dict generator doesn't cover the schedule key.

changing the config file `test-data/configuration_tests/ensemble_config.ert`, to include the line:

```
SCHEDULE_PREDICTION_FILE input/schedule.sch PARAMETER:input/param
```

and correspondingly updating the config dict in `tests/unit_tests/c_wrappers/res/enkf/test_ensemble_config.py`, line 95, to read

```
                   ConfigKeys.PARAMETER_KEY: "input/param",
```

gives us... drum roll... a util abort : )

Reply:
Did you also create that `param` file in `input/param` ? Also are you sure that it needs to have a particular format?

Reply:
yeah i now checked on main and saw that the util abort was that the file is missing. maybe that was the case on your branch as well, because no, i didn't create it. let me try again!

Reply:
so creating the file gives an exception, raised by the test, as it cannot create the representation of the parameter file:

```
/home/val/code/equinor/ert/src/ert/_c_wrappers/enkf/config/gen_kw_config.py:82: in getParameterFile:
...
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0xaf in position 0: invalid start byte
```

trying to fill the file with possibly meaningful content, like

```
WIGGLYNESS UNIFORM 0.98 1.03
```

gives the same error. this is on d0f4143fd, by the way, before your merged PR.

still trying to figure out if the parameter thingy is actually used in the context of a schedule prediction file. it seems to me like this was broken anyways...

Reply:
on current main, adding the changes described above, leads to the following value of `schedule_file_list` in ensemble config's `from_dict`:

```
[
  '/tmp/pytest-of-val/pytest-65/test_ensemble_config_construct0/test_data/input/schedule.sch', 
  'PARAMETER:input/param'
]
```
but then in `__init__` it tries to add each of these as a separate schedule file thingy, and it crashed on the second one, because it interprets the whole string as the first argument to schedule, where it expects a template file:

```
Error message: gen_kw_config_set_template_file: the template_file:PARAMETER:input/param does not exist - aborting.
```

so the handling of that schedule config thingy needs to either be covered by moar tests (all the tests wohoo) and fixed, or canned, because it's not necessary to have the other args (maybe we can ask a schedule prediction file expert tomorrow)

Reply:
I will fix it for now and update the test `test_ensemble_config_constructor` to cover the case where a parameter file is provided. 

I created an issue (https://github.com/equinor/ert/issues/4288) to investigate if the extra arguments should be allowed or not.

Reply:
great, thank you!